core/builder/src/main/java/io/quarkus/builder/Execution.java

```diff
@@ -56,12 +68,36 @@
             done = true;
     }
 
+    Map<StepInfo, Integer> orderTopologically(Set<StepInfo> startSteps) {
+        if (!(startSteps instanceof SortedSet)) {
+            throw new IllegalArgumentException("startSteps must be a SortedSet");
+        }
+
+        /* Kahn's algorithm */
+        final Map<StepInfo, AtomicInteger> indegrees = new HashMap<StepInfo, AtomicInteger>();
+        final Deque<StepInfo> queue = new ArrayDeque<>(startSteps);
+        final Map<StepInfo, Integer> result = new LinkedHashMap<StepInfo, Integer>();
```

Comment:
One more thing that could be a field on `StepInfo` instead.

Reply:
+1

Reply:
There is `BuildStepBuilder.indegree` for this now in aacdcf7